### ISSUE TITLE
Additional classical ML models

### ISSUE THREAD

**Issue by myerspat** [aims-umich/pyMAISE#64]
## Background

The next development phase includes the implementation of the following models:

- Gaussian Process [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessClassifier.html),
- Gradient Boosting [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html),
- [Ridge Regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html),
- [ElasticNet](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.ElasticNet.html),
- AdaBoost [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.AdaBoostRegressor.html) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.AdaBoostClassifier.html),
- Extra Trees [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesRegressor.html) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesClassifier.html),
- Stacking [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.StackingRegressor.html) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.StackingClassifier.html),
- Multi-output [Regressor](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputRegressor.html#sklearn.multioutput.MultiOutputRegressor) and [Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html).

All models follow the same implementation already implemented in `pyMAISE/methods/` for the classical models; however, stacking and multi-output models will require additional functionality as they take estimator(s) as an argument, not shared by the other classical models. The first and most basic approach to this is to implement the above models in the same fashion as in the classical models in `pyMAISE/methods/`; however, an alternative approach, similar to that done with the neural networks in `pyMAISE/methods/nn/`, is to template much of this to possibly accommodate all sklearn models that follow a similar argument pattern. This may also reduce redundant code.

## Tasks

- [ ] Add any relevant documentation

Implement the following models:

- [x] Gaussian Process Regressor and Classifier
- [x] Gradient Boosting Regressor and Classifier
- [x] Ridge Regression
- [x] ElasticNet
- [x] AdaBoost Regressor and Classifier
- [x] Extra Trees Regressor and Classifier
- [x] Stacking Regressor and Classifier
- [x] MultiOutput Regressor and Classifier

Add these models to

- [x] `pyMAISE.Tuner`

Add these models and rerun the following benchmarks:

- [x] MIT Reactor 
- [ ] Reactor Physics
- [ ] Fuel Performance
- [ ] Heat Conduction
- [ ] BWR Micro Core
- [ ] HTGR Micro-Core Quadrant Power
- [ ] NEACRP C1 Rod Ejection Accident
- [ ] Critical Heat Flux (CHF) Prediction

## Definition of Done

All the above tasks are done, the pull request passes all tests, and the pull request is reviewed.
